MaterialLib/MPL/Medium.h

```diff
@@ -0,0 +1,80 @@
+/**
+ * \author Norbert Grunwald
+ * \date   07.09.2017
+ *
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+#pragma once
+
+#include <memory>
+#include <vector>
+#include "BaseLib/ConfigTree.h"
+
+#include "Phase.h"
+#include "Property.h"
+
+namespace MaterialPropertyLib
+{
+/// This class is for material objects on the Medium scale.
+///
+/// A Medium consists of an arbitrarily long vector of phases and an array of
+/// properties.
+class Medium
```

Comment:
Can the class Medium be final or is it planned in another PR to derive from Medium?